Controller/NotifyHandler.php

```diff
@@ -116,12 +128,19 @@ private function _onNotificationSuccess(Response $oResponse, BackendService $oBa
         $oTransaction->wdoxidee_ordertransactions__action
             = new Field($oPayment->oxpayments__wdoxidee_transactionaction->value);
         $oTransaction->wdoxidee_ordertransactions__type = new Field($oResponse->getTransactionType());
-        $oTransaction->wdoxidee_ordertransactions__state = new Field($aData['transaction-state']);
         $oTransaction->wdoxidee_ordertransactions__amount = new Field($oResponse->getRequestedAmount()->getValue());
         $oTransaction->wdoxidee_ordertransactions__currency
             = new Field($oResponse->getRequestedAmount()->getCurrency());
         $oTransaction->wdoxidee_ordertransactions__responsexml = new Field(base64_encode($oResponse->getRawData()));
         $oTransaction->wdoxidee_ordertransactions__date = new Field($sConvertedTimestamp);
+
+        // set the transaction state to closed if no further operations are possible anymore
+        if ($oBackendService->isFinal($oResponse->getTransactionType())) {
```

Comment:
please without else set the transaction to success an make only a if its final the value is overridden with closed

Reply:
👍